pkg/v1/builder/command/cli_add.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package command provides handling to generate new scaffolding, compile, and
+// publish CLI plugins.
+package command
+
+import (
+	"errors"
+	"os"
+
+	"github.com/aunum/log"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/builder/template"
+)
+
+// looksLikeARepo tries to verify we are running from the root of a repo or
+// submodule. It looks for the presence of a go.mod and Makefile. If either is
+// not found, returns an error to indicate it does not look like we are running
+// from the repo root.
+func looksLikeARepo() error {
+	if _, err := os.Stat("go.mod"); errors.Is(err, os.ErrNotExist) {
+		return errors.New("command should be run from the root of the repo, go.mod not found")
+	}
+	if _, err := os.Stat("Makefile"); errors.Is(err, os.ErrNotExist) {
+		return errors.New("command should be run from the root of the repo, Makefile not found")
+	}
+	return nil
+}
+
+func AddPlugin(name, description string, dryRun bool) error {
```

Comment:
Missing godoc comment. Actually, I don't why the linter is not complaining.

Reply:
It didn't complain before either, so it's not related to where the code is in the tree since this was just moved as-is. I can add a comment though.